### PR TITLE
Cleanup Identifier mappings

### PR DESCRIPTION
**Pull request by ChloeDawn**

### CODE REVIEW

**Comment by liach** [comment]
err, why do you open a pr when we are updating to 1.14.1 pre 2

**Comment by ChloeDawn** [comment]
I didn't have Discord open so wasn't aware, I will retarget once Yarn is updated

**Comment by liach** [comment]
It has done updating less than 1 min ago.

File: mappings/net/minecraft/util/Identifier.mapping [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
-		METHOD deserialize (Lcom/google/gson/JsonElement;Ljava/lang/reflect/Type;Lcom/google/gson/JsonDeserializationContext;)Ljava/lang/Object;
-			ARG 1 functionJson
-			ARG 2 unused
-			ARG 3 context
 	FIELD a namespace Ljava/lang/String;
 	FIELD b path Ljava/lang/String;
-	FIELD c EXCEPTION_INVALID Lcom/mojang/brigadier/exceptions/SimpleCommandExceptionType;
-	METHOD <init> (Ljava/lang/String;)V
+	FIELD c COMMAND_EXCEPTION Lcom/mojang/brigadier/exceptions/SimpleCommandExceptionType;
+	METHOD <init> ([Ljava/lang/String;)V
```
Comment (NeunEinser) [inline mappings/net/minecraft/util/Identifier.mapping:6]: Is the parameter still `id` after the type changed to Array?

File: mappings/net/minecraft/util/Identifier.mapping [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
-	METHOD a parse (Lcom/mojang/brigadier/StringReader;)Lqs;
-	METHOD a create (Ljava/lang/String;)Lqs;
-		ARG 0 s
-	METHOD a createSplit (Ljava/lang/String;C)Lqs;
-		ARG 0 str
-		ARG 1 chr
+	METHOD a isCharValid (C)Z
+		ARG 0 c
+	METHOD a (I)Z
+		ARG 0 c
```
Comment (NeunEinser) [inline mappings/net/minecraft/util/Identifier.mapping:12]: What is this?

**Comment by asiekierka** [comment]
Is fine, this should be sane to rebase.

File: mappings/net/minecraft/util/Identifier.mapping [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
+		ARG 0 reader
+	METHOD a ofNullable (Ljava/lang/String;)Lqs;
+		ARG 0 id
+	METHOD a splitOn (Ljava/lang/String;C)Lqs;
+		ARG 0 id
+		ARG 1 delimiter
 	METHOD b getNamespace ()Ljava/lang/String;
-	METHOD b isValidIdentifier (Ljava/lang/String;)Z
+	METHOD b (I)Z
+		ARG 0 c
```
Comment (NeunEinser) [inline mappings/net/minecraft/util/Identifier.mapping:22]: And this?

File: mappings/net/minecraft/util/Identifier.mapping (hunk 2 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:12]: A `char` lambda parameter

File: mappings/net/minecraft/util/Identifier.mapping (hunk 1 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:6]: It's an array of components that form the `id`

File: mappings/net/minecraft/util/Identifier.mapping (hunk 3 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:22]: Same as other

**Comment by ChloeDawn** [comment]
Checked diff and should be good to merge directly as-is

File: mappings/net/minecraft/util/Identifier.mapping (hunk 2 above)
Comment (NeunEinser) [reply mappings/net/minecraft/util/Identifier.mapping:12]: I in the method signature is integer, not char.

File: mappings/net/minecraft/util/Identifier.mapping (hunk 3 above)
Comment (NeunEinser) [reply mappings/net/minecraft/util/Identifier.mapping:22]: It's also integer.

File: mappings/net/minecraft/util/Identifier.mapping (hunk 2 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:12]: It is a `char` represented as an `int` by an `IntConsumer` from the `IntStream` of `String#chars`: `String#chars().allMatch(c -> ...)`

**Review by NeunEinser — APPROVED** [review]

File: mappings/net/minecraft/util/Identifier.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/util/Identifier.mapping:12]: Is this a synthetic method generated for `METHOD a isCharValid (C)Z`?

Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:12]: No, the two `IntConsumer` synthetic lambda methods are for `isNamespaceValid` and `isPathValid`

File: mappings/net/minecraft/util/Identifier.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/Identifier.mapping:6]: `sanitizedParts`

File: mappings/net/minecraft/util/Identifier.mapping (hunk 2 above)
Comment (NeunEinser) [reply mappings/net/minecraft/util/Identifier.mapping:12]: We discussed it in discord. It's caused by String's `chars` method returning an `IntStream`.
Also, as Chloe mentioned before, it's a lambda.

File: mappings/net/minecraft/util/Identifier.mapping (hunk 1 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/util/Identifier.mapping:6]: The parts are not yet sanitized when passed into the constructor, it is merely a split string.

**Review by Gegy — APPROVED** [review]

**Review by liach — APPROVED** [review]
lgtm :+1:

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**